### ISSUE TITLE
MPI_ENABLED is not defined automatically when MPI is found

### ISSUE THREAD

**Issue by JoostHuizinga** [sferes2/sferes2#50]
When configuring a project on a machine that has both MPI and Boost installed, and both libraries are found successfully, the MPI_ENABLED flag is still not set. For example:

```
[jhuizing@mmmlog1 sferes2]$ ./waf configure --no-assert
WARNING simplejson not found some function may not work
Command-line options for module [modules/nn2] : OK 
Command-line options for module [modules/continue] : OK 
Command-line options for module [modules/fastsim2] : OK 
Command-line options for module [modules/randmaze] : OK 
Command-line options for module [modules/misc]     : OK 
Command-line options for exp [exp/map_elite]       :  -> OK 
Command-line options for exp [exp/example]         :  -> OK 
Command-line options for exp [exp/modularity]      :  -> OK 
Command-line options for exp [exp/cmoea]           :  -> OK 
Setting top to                           : /pfs/bh/home/jhuizing/sferes/sferes2 
Setting out to                           : /pfs/bh/home/jhuizing/sferes/sferes2/build 
Checking for 'g++' (C++ compiler)        : not found 
Checking for 'clang++' (C++ compiler)    : not found 
Checking for 'icpc' (C++ compiler)       : mpicxx 
### Checking for boost ###
Checking boost includes                  : 1_60 
Checking boost libs                      : ok 
Checking Intel TBB includes              : not found 
Checking for MPI include                 : ok 
Checking for Eigen                       : ok 
Configuring for module: [modules/nn2]
modules/nn2 -> ok
Configuring for module: [modules/continue]
modules/continue -> ok
Configuring for module: [modules/fastsim2]
Checking for SDL include                 : Not found 
modules/fastsim2 -> ok
Configuring for module: [modules/randmaze]
modules/randmaze -> ok
Configuring for module: [modules/misc]
modules/misc -> ok

--- configuration ---
compiler(s):
 * CXX: icc
boost version: 1_60
mpi: False
Compilation flags :
   CXXFLAGS : -D_REENTRANT -Wall -fPIC -ftemplate-depth-1024 -Wno-sign-compare -Wno-deprecated  -Wno-unused -DSFERES_ROOT="/pfs/bh/home/jhuizing/sferes/sferes2"  -std=c++11 -DNO_PARALLEL -DEIGEN3_ENABLED -DNDEBUG 
   LINKFLAGS: 
--- license ---
Sferes2 is distributed under the CECILL license (GPL-compatible)
Please check the accompagnying COPYING file or http://www.cecill.info/
'configure' finished successfully (1.243s)
```

Notice that the `-DMPI_ENABLED` flag is not part of the `CXXFLAGS`.

Looking at the code of the wscript in sferes root, I found that it fails on the following test:

```
    # boost mpi
    if (len(conf.env['LIB_BOOST_MPI']) != 0 and conf.env['MPI_FOUND']):
        print "MPI Enabled!"
        conf.env['MPI_ENABLED'] = True
    else:
        print "MPI Disabled! len(conf.env['LIB_BOOST_MPI'])", len(conf.env['LIB_BOOST_MPI']), "conf.env['MPI_FOUND']", conf.env['MPI_FOUND']
        conf.env['MPI_ENABLED'] = False
```

Which results in the following output (only the result of the print statement is shown):

```
MPI Disabled! len(conf.env['LIB_BOOST_MPI']) 0 conf.env['MPI_FOUND'] True
```

It turns out the `LIB_BOOST_MPI` environment property is never set. Looking at `boost.py` in `waf_tools`, shows why:

```
        var = kw.get('uselib_store', 'BOOST')

        self.start_msg('Checking boost includes')
        self.env['INCLUDES_%s' % var] = self.boost_get_includes(**params)
        self.env.BOOST_VERSION = self.boost_get_version(self.env['INCLUDES_%s' % var])
        self.end_msg(self.env.BOOST_VERSION)
        if Logs.verbose:
                Logs.pprint('CYAN', '   path : %s' % self.env['INCLUDES_%s' % var])

        if not params['lib']:
                return
        self.start_msg('Checking boost libs')
        suffix = params.get('static', 'ST') or ''
        path, libs = self.boost_get_libs(**params)
        self.env['%sLIBPATH_%s' % (suffix, var)] = [path]
        self.env['%sLIB_%s' % (suffix, var)] = libs
```

Here `var` is completely independent of the libraries found, meaning `LIB_BOOST` will be defined, but `LIB_BOOST_MPI` will not.

As for resolving this issue, we may be able to simply test for `not conf.options.no_mpi`, because `conf.check_boost(lib='mpi')` will halt configuration if it fails to find boost MPI. As it stands, unless `conf.options.no_mpi` is set, MPI is simply a requirement. (In fact, even if `conf.options.no_mpi` is set, sferes will still fail to configure if boost MPI is not found, but that is a different issue entirely #43).

**Comment by costashatz** [comment]
Did you try to fix this in #52? If not, I can give it a shot..

**Comment by costashatz** [comment]
Solved in #55 ...
